### PR TITLE
Deprecate confocal `{color}_image` properties

### PR DESCRIPTION
**Pull request by rpauszek**
**Why this PR?**
Currently you can access image data like `Kymo.green_image` however as more functionality is being added, it seems to make sense to have this as a method `.get_image()` with the color channel as the argument. Additionally, this unifies the API between `Scan` and `CorrelatedStack`, and simplifies some internal code slightly.

### CODE REVIEW

File: lumicks/pylake/detail/confocal.py [hunk 1]
```diff
@@ -321,7 +321,7 @@ def _get_plot_data(self, channel, adjustment=ColorAdjustment.nothing(), frame=No
         adjustment : lk.ColorAdjustment
             Color adjustments to apply to the output image if channel is set to "rgb".
         """
-        image = getattr(self, f"{channel}_image")
+        image = self.get_image(channel)  # getattr(self, f"{channel}_image")
```
Comment (tiagobonetti) [inline lumicks/pylake/detail/confocal.py:324]: Is this comment a leftover from the previous version?
The same comment is also present in lines 361, 369, and 407 in [test_calibrated_images.py](lumicks/pylake/kymotracker/tests/test_calibrated_images.py).

File: lumicks/pylake/tests/test_file/test_file_items.py [hunk 2]
```diff
@@ -78,24 +78,24 @@ def test_scans(h5_file):
...
 
 
 def test_kymos(h5_file):
     f = pylake.File.from_h5py(h5_file)
     if f.format_version == 2:
         kymo = f.kymos["Kymo1"]
         assert kymo.pixels_per_line == 5
-        np.testing.assert_allclose(kymo.red_image, np.transpose([[2, 0, 0, 0, 2], [0, 0, 0, 0, 0], [1, 0, 0, 0, 1], [0, 1, 1, 1, 0]]))
\ No newline at end of file
+        np.testing.assert_allclose(kymo.get_image("red"), np.transpose([[2, 0, 0, 0, 2], [0, 0, 0, 0, 0], [1, 0, 0, 0, 1], [0, 1, 1, 1, 0]]))
```
Comment (dean0x7d) [inline lumicks/pylake/tests/test_file/test_file_items.py:101]: :eolrip:

File: lumicks/pylake/tests/test_imaging_confocal/test_scan.py [hunk 3]
```diff
@@ -404,7 +414,7 @@ def test_plot_single_channel_percentile_color_adjustment(test_scans):
             scan.plot(
                 channel=channel, adjustment=ColorAdjustment(used_lb, used_ub, mode="percentile")
             )
-            image = getattr(scan, f"{channel}_image")
+            image = scan.get_image(channel) #getattr(scan, f"{channel}_image")
```
Comment (dean0x7d) [inline lumicks/pylake/tests/test_imaging_confocal/test_scan.py:417]: Leftover comment.

**Review by dean0x7d — COMMENTED** [review]
The code changes look good to me, but there are a couple of missing things:
* The docs need to be updated. Quite a few usages are in the docs.
* The Matlab examples also use this API. (We should consider dropping that half-baked Matlab support/examples.)

**Review by dean0x7d — APPROVED** [review]

**Pull request merged**